src/js/components/locations-configuration/LoadDataPage.jsx

```diff
@@ -90,31 +97,72 @@ const LoadDataContent = ({
       {ReactHtmlParser(summaryItemsList)}
     </div>
     <div className="d-flex justify-content-between m-3">
-      <button type="button" onClick={stepBackHandler} className="btn btn-outline-primary">
+      <button type="button" onClick={() => nextStep(null)} className="btn btn-outline-primary">
         <Translate id="default.button.back.label" defaultMessage="Back" />
       </button>
-      <button type="button" onClick={proceedHandler} className="btn btn-outline-primary">
+      <button type="button" onClick={() => nextStep(LOAD_DATA_STEPS.proceedLoadingDemoData)} className="btn btn-outline-primary">
         <Translate id="default.button.proceed.label" defaultMessage="Proceed" />
       </button>
     </div>
   </React.Fragment>
 );
 
+const LoadDataProgressScreen = ({ nextStep }) => {
+  useEffect(() => {
+    apiClient.get('/openboxes/api/config/data/demo')
+      .then(() => {
+        nextStep(LOAD_DATA_STEPS.dataLoadedSuccess);
```

Comment:
I don't like the idea of `dataLoadedSuccess` being its own step, it does not feel right to me. I'd prefer it to be content that is displayed in `then` and the same for error content being displayed in `catch`.